core/src/v3/openapi.rs

```diff
@@ -55,15 +55,46 @@ impl From<v2::DefaultApiRaw> for openapiv3::OpenAPI {
     }
 }
 
-fn openapi3_server(_host: Option<String>, base: Option<String>) -> Vec<openapiv3::Server> {
-    if let Some(base) = base {
+fn openapi3_servers(
+    schemes: std::collections::BTreeSet<v2::OperationProtocol>,
+    host: Option<String>,
+    base: Option<String>,
+) -> Vec<openapiv3::Server> {
+    if schemes.is_empty() && host.is_none() && base.is_none() {
+        vec![]
+    } else if let Some(host) = host {
+        if !schemes.is_empty() {
+            schemes
+                .into_iter()
+                .map(|scheme| {
+                    let scheme_str = match scheme {
+                        v2::OperationProtocol::Http => "http",
+                        v2::OperationProtocol::Https => "https",
+                        v2::OperationProtocol::Ws => "ws",
+                        v2::OperationProtocol::Wss => "wss",
+                    };
+                    openapiv3::Server {
+                        url: format!("{}://{}{}", scheme_str, host, base.as_deref().unwrap_or("")),
+                        description: None,
+                        variables: None,
+                        extensions: indexmap::IndexMap::new(),
+                    }
+                })
+                .collect()
+        } else {
+            vec![openapiv3::Server {
+                url: format!("//{}{}", host, base.as_deref().unwrap_or("")),
```

Comment:
Should this be:
`url: format!("{}{}", host, base.as_deref().unwrap_or("")),` ?

Reply:
Nope, if you do that way it will be a relative path while this URL (starting with //) uses the “current” protocol as per the v2 specification when the scheme is omitted